### ISSUE TITLE
crossEntropy does not validate that a target is inside the vocab

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#55]
Found while reviewing #54. Pre-existing and unchanged by that PR.

`crossEntropy` never checks that a target is inside the vocab:

```ts
total += Math.log(sum) + maxL - logits.data[b + targets[t]];
```

With `targets[t] >= V`, `b + targets[t]` reads into the next row's logits (or past the end of the array on the last row, giving `undefined` and a `NaN` loss). The result is a plausible-looking number rather than an error.

`softCrossEntropy` already validates, on the line right below:

```ts
if (id < 0 || id >= V) throw new Error(`softCrossEntropy: teacher id ${id} out of [0,${V})`);
```

so the two are inconsistent, and the one that does not check is the one on every training path.

Reachable by resuming a checkpoint against a corpus tokenized with a larger vocab, which is a real mistake to make: `agents.md` invariant 1 says the tokenizer freezes at step one precisely because that pairing is easy to get wrong. Today it trains on garbage instead of refusing.

The GPU path is safer by accident: `srcCeFwd` indexes `LOG[base + tgt]` where the buffer is bound with a fixed size, so a too-large target is clamped or discarded by WGSL's robust buffer access rather than reading a neighbouring row. That is another CPU/GPU divergence, though a benign-looking one.

The fix is the same one-line check `softCrossEntropy` already has, plus a case in `tests/gradcheck.ts`.
